File: src/main/java/com/example/demo/models/Equipment.java
package com.example.demo.models;

import com.example.demo.Enum.EquipmentStatus;
import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;

import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Entity
@Data
public class Equipment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String description;
    @Enumerated(EnumType.STRING)
    private EquipmentStatus status;

    @ManyToMany(mappedBy = "equipmentList") // Corrected mappedBy
    @JsonIgnore
    private Set<Incident> incidents = new HashSet<>();

    @OneToMany(mappedBy = "equipment")
    @JsonIgnore
    private List<Ticket> ticketList;
}

File: It_Angular/src/app/model/Equipment.ts
import {EquipmentStatus} from "../enum/EquipmentStatus";
import {Incident} from "./Incident";
import {Ticket} from "./Ticket";


export class Equipment {
  id!: number;
  name!: string;
  description!: string;
  status!: EquipmentStatus;
  incidents: Incident[] = [];
  ticketList: Ticket[] = [];
}
